promotion/src/main/java/org/cryptimeleon/incentive/promotion/PromotionReward.java

```diff
@@ -0,0 +1,12 @@
+package org.cryptimeleon.incentive.promotion;
+
+import lombok.Value;
+
+/**
+ * Represents what a user can gain from spending a certain amount of points.
+ */
+@Value
+public class PromotionReward {
+    long price;
```

Comment:
You interpret this value as the number of points this reward costs. But I feel like that's awkward, for example, for "buy 3, get one free" promotions. How many points does your free item cost?

Reply:
I guess it makes sense to describe the update that needs to happen when accepting this reward. 

Can we make this more general? Have some sort of `UpdateOperation` class that's associated with a reward and describes what must happen with the token when accepted? That operation would have to do some checks on the token content and then do some updates on it, afterwards. Basically describing what the ZK proof has to prove during Spend.

I feel like the current version probably will require larger reworks later on, so it may be worth it to already put the infrastructure in place to allow for fancier rewards.

Reply:
The equivalent to this `long price` would be an `UpdateOperation` that first checks that at least `price` points are present, then deducts `price` points.

Reply:
Okay, so you recommend building this more abstract such that we can easily include more complex operations once we add them to the ZKP, for example having an upper limit range proof specific to a reward, or rewards that require a certain amount of points but subtract another, e.g. a VIP status token that gets you 10% off but does not subtract points from your token?

Reply:
From my understanding this UpdateOperation is the part of a reward definition that determines the non-trusted input to 
the server side call of Spend-Deduct (https://github.com/cryptimeleon/incentive-system/blob/affd50c3b5bbf2deecba1e94a6f3b0442eb8c02f/crypto/src/main/java/org/cryptimeleon/incentive/crypto/IncentiveSystem.java#L390), do you agree?